services/galley/src/Galley/API/Query.hs

```diff
@@ -101,15 +101,18 @@ getConversation zusr cnv = do
   localDomain <- viewFederationDomain
   if qDomain cnv == localDomain
     then getUnqualifiedConversation zusr (qUnqualified cnv)
-    else getRemoteConversation zusr (toRemote cnv)
-
-getRemoteConversation :: UserId -> Remote ConvId -> Galley Public.Conversation
-getRemoteConversation zusr remoteConvId = do
-  conversations <- getRemoteConversations zusr [remoteConvId]
-  case conversations of
-    [] -> throwErrorDescription convNotFound
-    [conv] -> pure conv
-    _convs -> throwM (federationUnexpectedBody "expected one conversation, got multiple")
+    else getRemoteConversation (toRemote cnv)
+  where
+    getRemoteConversation :: Remote ConvId -> Galley Public.Conversation
+    getRemoteConversation remoteConvId = do
+      (foundConvs, _) <- Data.remoteConversationIdOf zusr [remoteConvId]
```

Comment:
There is a compile error here. Should it just be `foundConvs <-`?

Reply:
Ah yes, I refactored this. Note to self: it is seldom worth it to work on top of unmerged PR.